pyteal/ast/subroutine.py

```diff
@@ -1006,7 +1007,17 @@ def __call__(self, subroutine: SubroutineDefinition) -> SubroutineDeclaration:
             abi_output_kwargs[output_kwarg_info.name] = output_carrying_abi
 
         # Arg usage "B" supplied to build an AST from the user-defined PyTEAL function:
-        subroutine_body = subroutine.implementation(*loaded_args, **abi_output_kwargs)
+        subroutine_body: Expr
+        if not self.use_frame_pt:
+            subroutine_body = subroutine.implementation(
+                *loaded_args, **abi_output_kwargs
+            )
+        else:
+            with SubroutineEval.Context.CompileWithFrameContext(proto):
```

Comment:
Is this a lot cleaner than using a more direct approach that avoids python contexts? As cool as I find contexts to be, I think they're most useful for consumers of an API. In library code, they may cause unnecessary confusion.

Reply:
I see, let me see if there are some easy syntax to achieve same functionality of swapping scratch-based ABI locals to frame-based ABI locals. I was unaware of the purpose of context, and just make things easiest for me at this place.

Reply:
I figured out a way to get away with context syntax in this library internal implementation, pushed here: [7514b14](https://github.com/algorand/pyteal/pull/606/commits/7514b14397978e881e2354e014e4f9d5ca51cee6).